client/src/client.rs

```diff
@@ -769,47 +789,146 @@ impl Client {
                 HttpMethod::GET,
                 format!("{}/{}", &self.telemetry_url, uri::STATUS),
             )
-            .headers(self.headers.clone()),
+            .headers(self.headers.clone())
+            .body(None),
             StatusResponseHandler,
         )
     }
 }
 
+/// Events API flow initializer
+pub struct EventsApiFlowInit {
+    filter: EventFilter,
+    headers: HttpHeaders,
+    url: String,
+}
+
+impl EventsApiFlowInit {
+    /// Constructs new item with provided filter, headers and url
+    ///
+    /// # Errors
+    /// Fails if URL transformation fails
+    #[inline]
+    fn new<U>(filter: EventFilter, headers: HttpHeaders, url: U) -> Result<Self>
+    where
+        U: AsRef<str>,
+    {
+        Ok(Self {
+            filter,
+            headers,
+            url: transform_ws_url(url)?,
+        })
+    }
+}
+
+impl WebSocketFlowInit for EventsApiFlowInit {
+    type Next = EventsApiFlowHandshake;
+
+    fn init<B>(self) -> Result<(WebSocketInitData<B>, Self::Next)>
```

Comment:
`impl`, because the end user still sees the concrete types and *can* use the fact that it's the concrete implementation, encouraging non-portable use of the API.

Reply:
I'll try to expand on this a little. The function is generic in the `RequestBuilder`, but `RequestBuilder`s can be of different API behaviour. Right now there's only one implementer, but there could be more. Do you want your end user to be able to use the associated functions of that particular builder? 

I don't insist, but I think that hiding the implementation first using e.g. a marker trait `WebSocketInitDataTrait`, and changing the return type to `impl WebSocketInitDataTrait` can have its advantages. 

It's up to you, the generic return type can have its uses.